frontend/hub/collections/hooks/useCopyToRepository.tsx

```diff
@@ -0,0 +1,252 @@
+        const signingServiceName = props.context.settings.GALAXY_COLLECTION_SIGNING_SERVICE;
+
+        const url = pulpAPI`/signing-services/?name=${signingServiceName}`;
+        const signingServiceList = await pulpRequest(url);
+
+        signingService = signingServiceList?.results?.[0].pulp_href;
+      }
+
+      const repoHrefs: string[] = [];
+
+      for (const repo of selectedRepositories) {
+        if (!fixedRepositories.find((item) => item.name == repo.name)) {
+          repoHrefs.push(repo.pulp_href);
+        }
+      }
+
+      const params: {
+        collection_versions: string[];
+        destination_repositories: string[];
+        signing_service?: string;
+      } = {
+        collection_versions: collection.collection_version?.pulp_href
+          ? [collection.collection_version?.pulp_href]
+          : [],
+        destination_repositories: repoHrefs,
+      };
+
+      if (signingService) {
+        params.signing_service = signingService;
+      }
+
+      let api_op = operation as string;
+      if (operation == 'approve') {
+        api_op = 'move';
+      }
+
+      // TODO test errors
+      await hubAPIPost(
+        pulpAPI`/repositories/ansible/ansible/${pulpId || ''}/${api_op}_collection_version/`,
+        params
+      );
+
+      setIsLoading(false);
+      props.onClose();
+    } catch (error) {
+      setError(
+        operation == 'approve' ? t('Can not approve collection.') : t('Can not copy collection')
+      );
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    async function getSelected() {
+      const repos = await request(
+        hubAPI`/v3/plugin/ansible/search/collection-versions?limit=100000&name=${
+          collection.collection_version?.name || ''
+        } &&version=${collection.collection_version?.version || ''}`
+      );
+
+      if (repos.data?.length > 0) {
+        const reposMapped = repos.data.map((item) => item.repository || ({} as Repository));
+        setFixedRepositories(reposMapped);
+      }
+    }
+
+    void (async () => {
+      await getSelected();
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  let queryParams = undefined;
+  if (operation == 'approve') {
+    queryParams = { pulp_label_select: 'pipeline=approved' };
+  }
+
+  const view = usePulpView({
+    url: pulpAPI`/repositories/ansible/ansible/`,
+    toolbarFilters,
+    tableColumns,
+    disableQueryString: true,
+    keyFn: (item) => item.name,
+    queryParams,
+  });
+
+  return (
+    <Modal
+      title={t(`Select repositories`)}
+      aria-label={t(`Select repositories`)}
+      isOpen
+      onClose={() => {
+        props.onClose();
+      }}
+      variant={ModalVariant.large}
+      tabIndex={0}
+      actions={[
+        <Button
+          key="select"
+          variant="primary"
+          id="select"
+          onClick={() => {
+            void (async () => {
+              await copyToRepositories();
+            })();
+          }}
```

Comment:
```suggestion
          onClick={() => copyToRepositories()}
```

Reply:
This is because of error:  180:13  error  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

Reply:
so... `onClick={() => void copyToRepositories()}` ?

Reply:
Magic :-)

Reply:
But it does not work, the function is than not called, I reset this commit back.